daemons/controld/controld_fencing.c

```diff
@@ -242,42 +242,31 @@ send_stonith_update(const char *target, const char *uuid)
     }
 
     if (peer->xml_id == NULL) {
-        crm_info("Recording XML ID '%s' for node '%s'", uuid, target);
-        peer->xml_id = pcmk__str_copy(uuid);
+        crm_info("Recording XML ID '%s' for node '%s'", target_xml_id, target);
+        peer->xml_id = pcmk__str_copy(target_xml_id);
     }
 
     crmd_peer_down(peer, TRUE);
 
-    /* Generate a node state update for the CIB */
     node_state = create_node_state_update(peer, flags, NULL, __func__);
+    crm_xml_add(node_state, PCMK_XA_ID, target_xml_id);
 
-    /* we have to mark whether or not remote nodes have already been fenced */
     if (pcmk_is_set(peer->flags, pcmk__node_status_remote)) {
         char *now_s = pcmk__ttoa(time(NULL));
 
         crm_xml_add(node_state, PCMK__XA_NODE_FENCED, now_s);
         free(now_s);
     }
 
-    /* Force our known ID */
-    crm_xml_add(node_state, PCMK_XA_ID, uuid);
-
     rc = controld_globals.cib_conn->cmds->modify(controld_globals.cib_conn,
```

Comment:
would be good to error-check the result (making sure nonnegative, since `only_success == FALSE`) but not necessary for this PR

Reply:
Never mind, `cib_fencing_updated()` gets the error code and responds accordingly.

Reply:
:-/ The callback is even designed for only_success==TRUE

Not going to bother now though